core/src/main/java/org/elasticsearch/search/searchafter/SearchAfterBuilder.java

```diff
@@ -47,13 +45,34 @@
 /**
  *
  */
-public class SearchAfterBuilder implements ToXContent, FromXContentBuilder<SearchAfterBuilder>, Writeable<SearchAfterBuilder> {
-    public static final SearchAfterBuilder PROTOTYPE = new SearchAfterBuilder();
+public class SearchAfterBuilder implements ToXContent, Writeable<SearchAfterBuilder> {
     public static final ParseField SEARCH_AFTER = new ParseField("search_after");
     private static final Object[] EMPTY_SORT_VALUES = new Object[0];
 
     private Object[] sortValues = EMPTY_SORT_VALUES;
 
+    public SearchAfterBuilder() {
+    }
+
+    /**
+     * Read from a stream.
+     */
+    public SearchAfterBuilder(StreamInput in) throws IOException {
+        int size = in.readVInt();
+        sortValues = new Object[size];
+        for (int i = 0; i < size; i++) {
+            sortValues[i] = in.readGenericValue();
+        }
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeVInt(sortValues.length);
+        for (Object fieldValue : sortValues) {
+            out.writeGenericValue(fieldValue);
```

Comment:
So as far as I see is this is the crucial change in this PR? I was wondering if might have undesired effects that we allow more field value types to be serialized/deserialized than before by using `writeGenericValue`. What would happen for example when fieldValue is a GeoPoint. It would have caused the serialization to trip previously, now it will be okay (and I guess it might cause error later). 
I guess switching to `writeGenericValue` is a good tradeoff here but would like to hear your ideas about that.